internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -480,7 +482,8 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	if err := CheckRegionConfigsPriorityOrder(params.GetReplicationSpecs()); err != nil {
 		return diag.FromErr(err)
 	}
-	cluster, _, err := connV2.ClustersApi.CreateCluster(ctx, projectID, params).Execute()
+	// cannot call 2024-10-23 as it can enable ISS autoscaling when using old sharding configuration
+	cluster, _, err := connV220240805.ClustersApi.CreateCluster(ctx, projectID, params).Execute()
```

Comment:
`advanced_cluster` and `cluster` resources making use of `2024-08-05` for PATCH and POST operations will continue to do so. This avoids usage of `2024-10-23` and ensures no unintentional shard level autoscaling is triggered. 
Simplifying the amount of versions we use in advanced_cluster will be handled in coming autoscaling epic.